pyproject.toml

```diff
@@ -66,7 +65,8 @@ dependencies = [
   "isort>=5.12,<6",
   "black>=23.3,<24",
   "ruff==0.0.260",
-  "mypy>=1.3,<2",
+  "mypy==1.6.1",
+  "pydantic>=1.10,<2",             # The types in this codebase with mypy checker are only compatible with pydantic 1
```

Comment:
Hmmm I wonder if we should also restrict this in prod? Why only change dev?

Reply:
This is a problem only with mypy and the type annotations. This problem doesn't impact the production release version.